AnkiHubSoftware/ankihub_addon
# [BUILD-458] feat: (un)suspend notes from iframe

The AnkiHub AI chatbot will have buttons that will allow users to unsuspend notes (or suspend them). This task is for implementing the add-on side of this.

The AnkiHub AI templates will be able to use java script like this to send a message to python:
```js
const message = {
    // The format of the message is "<command> <args_dict>"
    message: `ankihub_unsuspend_notes {
      "noteIds": [
        "00000000-0000-0000-0000-000000000001", "00000000-0000-0000-0000-000000000002"
      ]
    }`,
    sendToPython: true
};
window.parent.postMessage(message, '*');
```

This will work analogously for `ankihub_suspend_notes`.

## Related issues
https://ankihub.atlassian.net/jira/software/c/projects/BUILD/boards/1?selectedIssue=BUILD-458

## Proposed changes
- Move `unsuspend_notes` function from `gui.deck_updater.AnkiHubDeckUpdater` to `gui.operations.scheduling` because it will be used in multiple places now.
- Add a `suspend_notes` function to `gui.operations.scheduling`
- Handle `ankihub_suspend_notes` and `ankihub_unsuspend_notes` pycmds by calling `suspend_notes` or `unsuspend_notes`.

## How to reproduce
- Comment out this code to make the chatbot show up for any AnkiHub deck:
  https://github.com/ankipalace/ankihub_addon/blob/1a8b0eb4b3c2b030b0c538c44fea8fc6014e151f/ankihub/gui/reviewer.py#L104-L107
- Add this code:
  ```js
  const message = {
    message: `ankihub_suspend_notes {"noteIds": [
      // Change these to AnkiHub note ids of notes that you have in your Anki collection
      "59d5f5c4-2d52-413b-8895-5999c702200a", "d3ec9513-b4d6-43ab-b5dc-3a9189a93d61"
    ]}`,
    sendToPython: true
  }
  window.parent.postMessage(message, '*')
  ```
  to the web app `chatbot.html` template below this:
  https://github.com/ankipalace/ankihub/blob/f6c9d7fce05f46a03fcdce86b5f624b449496a1c/ankihub/templates/ai/chatbot.html#L92-L93
  This way the `ankihub_suspend_notes` message will be sent when you send a message to the chatbot. (This is just for testing and not how it will work for real.)
- Enable the `chatbot` feature flag on your local ankihub web app
- Open an ankihub note in the reviewer
- Send a message to the chatbot
- The notes with the provided ids should now be suspended
- You can try the same thing with `ankihub_unsuspend_notes`
